pkg/resource/certificate_authority_activation/hooks.go

```diff
@@ -0,0 +1,126 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package certificate_authority_activation
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	ackerr "github.com/aws-controllers-k8s/runtime/pkg/errors"
+	ackrtlog "github.com/aws-controllers-k8s/runtime/pkg/runtime/log"
+	svcsdk "github.com/aws/aws-sdk-go/service/acmpca"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+)
+
+var (
+	mu sync.Mutex
+)
+
+func (rm *resourceManager) customFindCertificateAuthorityActivation(
+	ctx context.Context,
+	r *resource,
+) (latest *resource, err error) {
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.customFindCertificateAuthorityActivation")
+	defer func() {
+		exit(err)
+	}()
+
+	// If any required fields in the input shape are missing, AWS resource is
+	// not created yet. Return NotFound here to indicate to callers that the
+	// resource isn't yet created.
+	if r.ko.Spec.CertificateAuthorityARN == nil {
+		return nil, ackerr.NotFound
+	}
+
+	// lock runtime
+	mu.Lock()
+	defer mu.Unlock()
```

Comment:
Thoughts: I was wondering we should use a internal cache instead of make api server requests everytime... definetly something to consider before going GA